apps/extension/src/shared/components/numeraires-form.tsx

```diff
@@ -13,39 +12,36 @@ const useNumerairesSelector = (state: AllSlices) => {
     selectedNumeraires: state.numeraires.selectedNumeraires,
     selectNumeraire: state.numeraires.selectNumeraire,
     saveNumeraires: state.numeraires.saveNumeraires,
-    networkChainId: state.network.chainId,
   };
 };
 
 export const NumeraireForm = ({
   isOnboarding,
   onSuccess,
+  chainId,
 }: {
+  chainId?: string;
   isOnboarding?: boolean;
-  onSuccess: () => void | Promise<void>;
+  onSuccess: () => void;
 }) => {
-  const { chainId } = useChainIdQuery();
-  const { selectedNumeraires, selectNumeraire, saveNumeraires, networkChainId } =
-    useStore(useNumerairesSelector);
-
-  // 'chainId' from 'useChainIdQuery' is not available during onboarding,
-  // this forces you to use two sources to guarantee 'chainId' for both settings and onboarding
-  const { numeraires } = useNumeraires(chainId ?? networkChainId);
+  const { selectedNumeraires, selectNumeraire, saveNumeraires } = useStore(useNumerairesSelector);
+  const { numeraires, isLoading, isError } = useNumeraires(chainId);
 
+  // If query errors or there aren't numeraires in the registry, can skip
   useEffect(() => {
-    if (numeraires.length === 0) {
-      void onSuccess();
+    if (isError || (!isLoading && numeraires.length === 0)) {
+      onSuccess();
     }
-  }, [numeraires]);
+  }, [numeraires.length, isError, isLoading]);
```

Comment:
The previous source of the bug. During the loading state, the numeraires array was length zero. Hence it was being skipped. Now it only skips if there is an error or it's still loading.